cloudinit/config/cc_growpart.py

```diff
@@ -305,14 +305,7 @@ def device_part_info(devpath):
         # FreeBSD doesn't know of sysfs so just get everything we need from
         # the device, like /dev/vtbd0p2.
         fpart = "/dev/" + util.find_freebsd_part(devpath)
-        m = re.search("^(/dev/.+)p([0-9])$", fpart)
-        if m:
-            return m.group(1), m.group(2)
-
-        # alternatively, we could be dealing with MBR slices,
-        # or we're on DragonFly
-        fslice = "/dev/" + util.find_dragonflybsd_part(devpath)
-        m = re.search("^(/dev/.+)s([0-9])$", fslice)
+        m = re.search("^(/dev/.+)[sp]([0-9]+[a-z]*)$", fpart)
```

Comment:
@igalic Let's leave some comment around the slices vs partitions as future-me won't recall why we have the 's' match.
`# We could be dealing with partition devices 'p' or MBR slices 's'.

Also we've added [a-z] supplements here to the match. This now permits /dev/<anything>s9zzzzzzzzzasdfzzz. What device types require this supplemental slicing id suffix? Can we add a comment  or unit test that exercises the expected /dev/ examples which meet these new matches?

Also since we are touching this regex might it be worth naming the symbolic group matches so we don't have discern below m.group(1) m.group(2)?
```suggestion
        m = re.search("^(?P<dev>/dev/.+)[sp](?P<part_slice>[0-9]+[a-z]*)$", fpart)
        if m:
            return m["dev"], m['part_slice"]
```